core/src/mindustry/ui/SearchBar.java

```diff
@@ -0,0 +1,49 @@
+package mindustry.ui;
+
+import arc.func.*;
+import arc.scene.ui.layout.*;
+import arc.struct.*;
+import mindustry.gen.*;
+
+public class SearchBar{
+    public static <T> Table add(Table parent, Seq<T> list, Func<String, String> queryf,
+            Func<T, String> namef, Cons2<Table, T> itemf){
+        Cons<String>[] rebuild = new Cons[] {null};
+        Table[] pane = new Table[] {null};
```

Comment:
Anuke hate spaces :eyes:

```suggestion
        Cons<String>[] rebuild = new Cons[]{null};
        Table[] pane = new Table[]{null};
```